src/summary.rs

```diff
@@ -305,62 +307,90 @@ mod tests {
             .unwrap();
         let some_text = " bla \t\n\tbla\n\t".to_string();
 
+        async fn write_file_to_blobdir(d: &TestContext) -> PathBuf {
+            let bytes = &[38, 209, 39, 29]; // Just some random bytes
+            let file = d.get_blobdir().join("random_filename_392438");
+            tokio::fs::write(&file, bytes).await.unwrap();
+            file
+        }
+
         let msg = Message::new_text(some_text.to_string());
         assert_summary_texts(&msg, ctx, "bla bla").await; // for simple text, the type is not added to the summary
 
+        let file = write_file_to_blobdir(&d).await;
         let mut msg = Message::new(Viewtype::Image);
-        msg.set_file("foo.jpg", None);
+        msg.set_file_and_deduplicate(&d, &file, Some("foo.jpg"), None)
+            .unwrap();
         assert_summary_texts(&msg, ctx, "📷 Image").await; // file names are not added for images
 
+        let file = write_file_to_blobdir(&d).await;
         let mut msg = Message::new(Viewtype::Image);
         msg.set_text(some_text.to_string());
-        msg.set_file("foo.jpg", None);
+        msg.set_file_and_deduplicate(&d, &file, Some("foo.jpg"), None)
+            .unwrap();
         assert_summary_texts(&msg, ctx, "📷 bla bla").await; // type is visible by emoji if text is set
 
+        let file = write_file_to_blobdir(&d).await;
         let mut msg = Message::new(Viewtype::Video);
-        msg.set_file("foo.mp4", None);
+        msg.set_file_and_deduplicate(&d, &file, Some("foo.mp4"), None)
+            .unwrap();
         assert_summary_texts(&msg, ctx, "🎥 Video").await; // file names are not added for videos
 
+        let file = write_file_to_blobdir(&d).await;
```

Comment:
Why do you write the same file over and over again here?

Reply:
Because `set_file_and_deduplicate()` will move the file away

Reply:
Yeah I find it a bit excessive to create so many files while it might in theory be enough to just move one file around but in the end it does not really matter.